pkg/virt-handler/vm.go

```diff
@@ -1422,12 +1417,53 @@ func (d *VirtualMachineController) calculateLiveMigrationCondition(vmi *v1.Virtu
 		return newNonMigratableCondition("VMI uses virtiofs", v1.VirtualMachineInstanceReasonVirtIOFSNotMigratable), isBlockMigration
 	}
 
+	containsPCIHostDev, err := vmiContainsPCIHostDevice(vmi)
+	// should never happen; checking error in case the underline implementation will be changed
+	if err != nil {
+		return newNonMigratableCondition("can't get host devices", v1.VirtualMachineInstanceReasonHostDeviceNotMigratable), isBlockMigration
+	}
+
+	if containsPCIHostDev {
+		return newNonMigratableCondition("VMI uses a PCI host devices", v1.VirtualMachineInstanceReasonHostDeviceNotMigratable), isBlockMigration
+	}
+
 	return &v1.VirtualMachineInstanceCondition{
 		Type:   v1.VirtualMachineInstanceIsMigratable,
 		Status: k8sv1.ConditionTrue,
 	}, isBlockMigration
 }
 
+// Note: genhostdev.CreateHostDevices uses NewBestEffortAddressPool, that its Pop() function does not return error; the
+// meaning is that this function won't return error unless the implementation of the genhostdev.CreateHostDevices function
+// will be changed.
+func vmiContainsPCIHostDevice(vmi *v1.VirtualMachineInstance) (bool, error) {
+	if len(vmi.Spec.Domain.Devices.HostDevices) > 0 {
+		hostDevs, err := genhostdev.CreateHostDevices(vmi.Spec.Domain.Devices.HostDevices)
```

Comment:
I don't think this will work the way you expect it... 
Normally, this function would be called from within the virt-launcher where you would find an environment variable (or many), representing an allocated resource(s), with a resource ID 
Virt-handler will not have that... therefore, i doubt this works.

Reply:
An alternative approach is to examine the domain XML of the created VMI directly using `domain.GetXMLDesc`
However, I think this is not needed at this point.

Reply:
Thanks @vladikr - done